src/wallet/bdk.rs

```diff
@@ -103,11 +103,40 @@ impl KeychainKind {
         match self {
             Self::External => 0,
             Self::Internal => 1,
-            Self::Fidelity => 2,
-            Self::SwapCoin => 3,
-            Self::Contract => 4,
+            Self::Fidelity { .. } => 2,
+            Self::SwapCoin { .. } => 3,
+            Self::Contract { .. } => 4,
         }
     }
+
+    /// return Fidelity index no
+    pub fn fidelity_kind_index() -> u32 {
+        KeychainKind::Fidelity { count: 0 }.index_num()
+    }
+
+    /// returns Contract index no
+    pub fn contract_kind_index() -> u32 {
+        KeychainKind::Contract { count: 0 }.index_num()
+    }
```

Comment:
We 3 major APIs at this stage.

- Add a new keychain index with a given Xpriv derived descriptor. This should take `&mut Wallet` and mutate the signer container and indexed_tx_graph.
- Get the derived Xpub descriptor at a given keychain index. If no index found return a None.
- Get the derived Keymap given a keychain index. Return None if not found.

With this in place you should be able to write basic unit tests.

- get a random Xpriv desriptor. Add it in a custom keychain.
- Get the Xpub and Xpriv via APIs.
- Assert the `xpub == xpriv.derive_publickey()`.
- Try that with one test vector of all the types of descriptors we are concerned here. wsh, wsh(timelock), wsh(htlc), wsh(multisig).

With this in place, writing init and load function should be easy.

Reply:
Yes I meant Wallet API. The Keychain enum doesn't need anything (for now) other than the .path() method.